test/support/test_case.rb

```diff
@@ -75,334 +79,108 @@ def program_with_line_numbers(program)
       lines_with_number.join("\n")
     end
 
-    def create_protocol_message fail_msg
-      all_protocol_msg = <<~DEBUGGER_MSG.chomp
-        -------------------------
-        | All Protocol Messages |
-        -------------------------
-
-        #{@backlog.join("\n")}
-      DEBUGGER_MSG
-
-      case ENV['RUBY_DEBUG_TEST_UI']
-      when 'vscode'
-        pattern = /(D|V)(<|>)(D|V)\s/
-      when 'chrome'
-        pattern = /(D|C)(<|>)(D|C)\s/
-      end
-
-      last_msg = @backlog.last(3).map{|log|
-        json = log.sub(pattern, '')
-        JSON.pretty_generate(JSON.parse json)
-      }.join("\n")
-
-      last_protocol_msg = <<~DEBUGGER_MSG.chomp
-        --------------------------
-        | Last Protocol Messages |
-        --------------------------
-
-        #{last_msg}
-      DEBUGGER_MSG
-
-      debuggee_msg =
-          <<~DEBUGGEE_MSG.chomp
-            --------------------
-            | Debuggee Session |
-            --------------------
-
-            > #{@remote_info.debuggee_backlog.join('> ')}
-          DEBUGGEE_MSG
-
-      failure_msg = <<~FAILURE_MSG.chomp
-        -------------------
-        | Failure Message |
-        -------------------
-
-        #{fail_msg}
-      FAILURE_MSG
-
-      <<~MSG.chomp
-        #{all_protocol_msg}
+    def type(command)
+      @scenario.push(command)
+    end
 
-        #{last_protocol_msg}
+    def multithreaded_test?
+      Thread.current[:is_subthread]
+    end
 
-        #{debuggee_msg}
+    ASK_CMD = %w[quit q delete del kill undisplay].freeze
 
-        #{failure_msg}
-      MSG
+    def debug_print msg
+      print msg if ENV['RUBY_DEBUG_TEST_DEBUG_MODE']
     end
-  end
 
-  class WebSocketClient
-    class Frame
-      attr_reader :binary
+    RUBY = ENV['RUBY'] || RbConfig.ruby
+    RDBG_EXECUTABLE = "#{RUBY} #{__dir__}/../../exe/rdbg"
 
-      def initialize
-        @binary = ''.b
-      end
+    TIMEOUT_SEC = (ENV['RUBY_DEBUG_TIMEOUT_SEC'] || 10).to_i
 
-      def << obj
-        case obj
-        when String
-          @binary << obj.b
-        when Enumerable
-          obj.each{|e| self << e}
-        end
-      end
+    private
 
-      def char bytes
-        @binary << bytes
-      end
+    def wait_pid pid, sec
+      total_sec = 0.0
+      wait_sec = 0.001 # 0.1ms
```

Comment:
This is 1ms right? Not 0.1ms.

Reply:
Yeah. But in order to preserve the behavior, I'll update the comment instead.